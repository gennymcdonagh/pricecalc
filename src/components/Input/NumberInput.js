import React, { useState } from 'react';
import './Input.css';
import PropTypes from 'prop-types';

export const NumberInput = (props) => {
  const {id, label, value, onChange, error} = props;
  const [showError, setShowError] = useState(false);

  return (
    <div className="number-input">
      <label htmlFor={id}>
        {label}
      </label>
      <input
          value={value}
          onChange={e => {setShowError(false); onChange(e);}}
          onBlur={() => error && setShowError(true)}
          id={id}
          type="number"
          name={id}
          required
        />
        {showError && <div className="number-input__error">{error}</div>}
    </div>
  )
};

NumberInput.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  error: PropTypes.string
}

