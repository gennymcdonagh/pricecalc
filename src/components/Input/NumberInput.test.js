import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { NumberInput } from './NumberInput';

it('renders correctly', () => {
  const numberInput = renderer.create(<NumberInput/>).toJSON();
  expect(numberInput).toMatchSnapshot();
})
