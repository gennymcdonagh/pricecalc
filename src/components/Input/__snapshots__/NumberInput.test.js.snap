// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="number-input"
>
  <label
    className="number-input__label"
  />
  <div
    className="number-input__wrapper"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
</div>
`;
